File: Week 6/sentimental-readability/readability.py

def main():
    # get string from user
    text = input("Text: ")
    letter_counter, word_counter, sentence_counter = string_counts(text)
    # define grading index
    index = round((0.0588 * ((letter_counter * 100) / word_counter)) - (0.296 * ((sentence_counter * 100) / word_counter)) - 15.8)

    # print grade based on index
    if 0 < index <= 16:
        print(f"Grade {index}")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade 16+")
    return 0


def string_counts(a):
    letter_counter = 0
    word_counter = 0
    sentence_counter = 0
    # loop through phrases counting letters words and sentences
    for i in range(len(a)):
        # Count letter if alpha
        if a[i].isalpha():
            letter_counter += 1
        # assess if end of string
        elif i == len(a) - 1:
            # if ends with quotation only count word
            if a[i] == '"':
                word_counter += 1
            # else must be end of word and sentence
            else:
                word_counter += 1
                sentence_counter += 1
        # count spaces as words
        elif a[i].isspace():
            word_counter += 1
        # count punctuation as sentences
        elif a[i] == '.' or a[i] == '!' or a[i] == '?':
            sentence_counter += 1
    return letter_counter, word_counter, sentence_counter


if __name__ == "__main__":
    main()

File: Week 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string a);
int count_words(string b);
int count_sentences(string c);

int main(void)
{
    //get string from user
    string text = get_string("Text: ");

    //count number of letters in string
    int letter_count = count_letters(text);

    //count number of words
    int word_count = count_words(text);

    //count number of sentences
    int sentence_count = count_sentences(text);

    //convert ints to floats
    float letter_count_f = (float) letter_count;
    float word_count_f = (float) word_count;
    float sentence_count_f = (float) sentence_count;

    //define grading index
    float L = ((letter_count_f * 100) / word_count_f);
    float S = ((sentence_count_f * 100) / word_count_f);
    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    //print grade level
    if (index <= 16 && index > 0)
    {
        printf("Grade %d\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
    return 0;
}
int count_letters(string a)
{
    //loop through phrase counting words
    int counter = 0;
    for (int i = 0, n = strlen(a); i < n; i++)
    {
        if (isalpha(a[i]))
        {
            counter++;
        }
    }
    return counter;
}
int count_words(string b)
{
    //loop through phrase counting words
    int counter = 0;
    for (int i = 0, n = strlen(b); i <= n; i++)
    {
        if (isspace(b[i]) || b[i] == '\0')
        {
            counter++;
        }
    }
    return counter;
}
int count_sentences(string c)
{
    //loop through phrase counting sentences
    int counter = 0;
    for (int i = 0, n = strlen(c); i <= n; i++)
    {
        if ((c[i]) == '.' || c[i] == '!' || c[i] == '?')
        {
            counter++;
        }
    }
    return counter;
}